x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java

```diff
@@ -185,7 +186,7 @@ protected static QueryList termQueryList(
         return switch (inputDataType) {
             case IP -> QueryList.ipTermQueryList(field, searchExecutionContext, (BytesRefBlock) block);
             case DATETIME -> QueryList.dateTermQueryList(field, searchExecutionContext, (LongBlock) block);
-            default -> QueryList.rawTermQueryList(field, searchExecutionContext, block);
+            case null, default -> QueryList.rawTermQueryList(field, searchExecutionContext, block);
```

Comment:
`default` doesn't include nulls for enums (As enums call `v.ordinal()` in switches), and it throws unless an explicit null case is added.
Another interesting thing, is that the `case null, default` syntax is the only way to combine null or default cases!